x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/transform/action/PreviewTransformAction.java

```diff
@@ -52,19 +52,38 @@ private PreviewTransformAction() {
 
     public static class Request extends AcknowledgedRequest<Request> implements ToXContentObject {
 
+        private final String transformId;
         private final TransformConfig config;
 
+        public Request(String transformId) {
+            this.transformId = transformId;
+            this.config = null;
+        }
+
         public Request(TransformConfig config) {
+            this.transformId = null;
             this.config = config;
         }
 
         public Request(StreamInput in) throws IOException {
             super(in);
-            this.config = new TransformConfig(in);
+            if (in.getVersion().onOrAfter(Version.V_8_0_0)) {
+                this.transformId = in.readOptionalString();
+                this.config = in.readOptionalWriteable(TransformConfig::new);
+            } else {
+                this.transformId = null;
+                this.config = new TransformConfig(in);
+            }
         }
 
         public static Request fromXContent(final XContentParser parser) throws IOException {
             Map<String, Object> content = parser.map();
+            if (content.size() == 1 && content.containsKey(TransformField.ID.getPreferredName())) {
+                // The request only contains transform id so instead of parsing TransformConfig (which is pointless in this case),
+                // let's just fetch the transform config by id later on.
+                String transformId = (String) content.get(TransformField.ID.getPreferredName());
+                return new Request(transformId);
+            }
```

Comment:
This seems really strange to me. It should be:

 - Full config in the body of the request
 - OR just the ID in the API.

I am not sure all this special handling in the transport action is even required as the REST layer could do the simple action of grabbing the transform via the ID if they hit that API endpoint.

Reply:
BTW, I've just noticed there is no HLRC method for Data Frame Analytics preview. Is it expected?

Reply:
@przemekwitek there is probably no HLRC in 8.0.0 as the HLRC is not supported in 8.0. I am thinking its in 7.x.